pkg/maps/lbmap/ipv4.go

```diff
@@ -328,8 +328,13 @@ func (s *Service4Value) GetFlags() uint16 {
 func (s *Service4Value) SetSessionAffinityTimeoutSec(t uint32) {
 	// Go doesn't support union types, so we use BackendID to access the
 	// lb4_service.affinity_timeout field. Also, for the master entry the
-	// LB algorithm was set beforehand, so we need to binary OR here.
-	s.BackendID |= t
+	// LB algorithm can be set independently, so we need to preseve the
```

Comment:
I was not aware of https://github.com/cilium/cilium/pull/36204/, but I think it might make sense to change the method to return an error if a given affinity timeout `t` is > `0x00ffffff`. Then the error should be logged with level=warn. Otherwise, we might silently configure a different than expected by a user timeout.

Reply:
> I think it might make sense to change the method to return an error if a given affinity timeout `t` is > `0x00ffffff`.

Done. 

> Then the error should be logged with level=warn.

Errors were already propagated in `pkg/loadbalancer/experimental` and finally logged at `warn`, but error handling was more complicated in the non-experimental case (with call stack reaching for instance `api/v1/server/restapi/service/put_service_id.go` and `pkg/service/health_check.go` without logging), so as you suggested I added logging in `pkg/maps/lbmap/lbmap.go`.